common/str-base.cpp

```diff
@@ -850,6 +852,11 @@ TEMPLATE uint BASESTRING::hash() const {
 	return hashResult ^ _size;
 }
 
+template<>
```

Comment:
If I am not mistaken, this should be declared in str-base.h or at least there should be a definition.

From [cppreference.com](https://en.cppreference.com/w/cpp/language/template_specialization):
> Specialization must be declared before the first use that would cause implicit instantiation, in every translation unit where such use occurs

Reply:
I don't think that's a problem problem because BaseString isn't implicitly instantiated anywhere, it's only explicitly instantiated at the end of str-base.cpp, which is after the new defs.

Reply:
OK, that's true.